partner_relations/model/res_partner_relation_all.py

```diff
@@ -209,4 +210,6 @@ def create(self, vals):
     @api.one
     def unlink(self):
         """divert non-problematic creates to underlying table"""
-        return self.get_underlying_object().unlink()
+        underlying_objs = self.get_underlying_object()
+        underlying_objs = underlying_objs and underlying_objs[0]
+        return underlying_objs.unlink()
```

Comment:
Don't hesitate to correct me if i'm wrong but ```self.get_underlying_object()``` is call in an @api.one method.
Then if ```get_underlying_object``` use ```self.id```. This method should return a list with only 1 object.